lerp_derive/src/derive.rs

```diff
@@ -1,60 +1,212 @@
 use proc_macro2::TokenStream;
-use quote::{quote};
-use syn::{Error, Field, Fields, Index, ItemStruct, Result, spanned::Spanned};
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_quote,
+    punctuated::Punctuated,
+    spanned::Spanned,
+    Attribute, Error, Field, Fields, Index, ItemStruct, Path, Result, Token, Type, TypeGroup,
+    TypeParen, TypePath, TypeReference,
+};
 
-pub fn lerp_derive_internal(input: ItemStruct) -> Result<TokenStream> {
+enum LerpType {
+    Type(TypePath),
+    Generic,
+    Skip,
+}
+
+#[derive(Default)]
+struct LerpAttributes {
+    skip: Option<()>,
```

Comment:
Why `Option<()>` instead of `bool`?

Reply:
It was left over from a previous idea I had, ill make it a bool

Reply:
Also, how should we handle the types in arrays? allow for nesting or just 1 level of lerping.

Like should 
```[[f32; 10]; 10]```
be accepted or rejected

Reply:
Actually, now that I think of it, it'd be simplest to just add a generic

```rust
impl<T, F, N> Lerp<F> for [T; N] { ... }
impl<T, F> Lerp<F> for Vec<T> { ... }
```

and then handle those cases with the generic impl, instead of implementing those cases within the macro.

Reply:
Yah, looking into it that would be best.

Reply:
Getting a derive working for arrays is a mess

Reply:
Adding that derive is a bit hard since the error pops up
```
error[E0119]: conflicting implementations of trait `Lerp<_>` for type `std::vec::Vec<_>`:
   --> src/lib.rs:177:1
    |
166 | / impl<T, F> Lerp<F> for T
167 | | where
168 | |     T: Add<Output = T> + Mul<F, Output = T>,
169 | |     F: Float,
...   |
173 | |     }
174 | | }
    | |_- first implementation here
...
177 |   impl<T: Lerp<F>, F: Float> Lerp<F> for Vec<T> {
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `std::vec::Vec<_>`
    |
    = note: upstream crates may add a new impl of trait `std::ops::Add` for type `std::vec::Vec<_>` in future versions
    = note: downstream crates may implement trait `std::ops::Mul<_>` for type `std::vec::Vec<_>`
    = note: downstream crates may implement trait `std::ops::Mul<_>` for type `std::vec::Vec<_>`
```

This Is hard to get around, without making the previous blanket implementation less general.

Reply:
Agree. I don't think it's worth stalling this PR any more for that. For now, I think the way forward is to 

- create a new issue in this repo that Lerp derives don't work for vectors or arrays yet
- add a comment in some appropriate place linking to that issue
- just don't handle arrays or vectors in this PR at all

Does that work for you?

Reply:
Indeed, Ill do that now